src/tool_getparam.c

```diff
@@ -2082,22 +2083,22 @@ ParameterError getparameter(const char *flag, /* f or -long-flag */
         break;
       }
       now = time(NULL);
-      config->condtime = curl_getdate(nextarg, &now);
-      if(-1 == (int)config->condtime) {
+      config->condtime = (long)curl_getdate(nextarg, &now);
```

Comment:
though technically getdate can't handle beyond 2038 despite returning time_t and therefore this cast is fine it _does_ return a time_t so best practice would be check for overflow , example 
{time_t d = curl_getdate(nextarg, &now); if(d>LONG_MAX) {warnf(global, "Time overflow for -z, year >2038 not supported"); config->condtime = CURL_TIMECOND_NONE;} else config->condtime = (long)d;}

Reply:
just for clarification: *getdate() can handle > 2038 dates fine if `time_t` is larger than 32 bits. ..

Reply:
> just for clarification: *getdate() can handle > 2038 dates fine if time_t is larger than 32 bits. ..

but does it,
https://github.com/curl/curl/blob/62c07b5743490ce373910f469abc8cdc759bec2b/lib/parsedate.c#L529-L541

in windows `long` is 32 bits. since there is no portable MAX time_t we could do use CURL_OFF_T_MAX since just about any recent compiler should size curl_off_t largest

Reply:
That looks like a bug on (at least) win64 that we should fix! IIRC, `time_t` is 64 bit there.

Reply:
The time values for ``CURLINFO_FILETIME`` and ``CURLOPT_FILETIME`` are currently limited to 32 bits on Windows, see https://github.com/curl/curl/pull/1378#issuecomment-291017171

I will add the necessary overflow checks.

Reply:
Maybe `lib/smb.c:get_posix_time()` (that I'm polishing slightly in #2240) could be generalized and used for this purpose here as well ?